test/cppsuite/src/main/database.cpp

```diff
@@ -72,11 +91,15 @@ database::get_collection(uint64_t id)
 collection &
 database::get_random_collection()
 {
+    std::lock_guard<std::mutex> lg(_mtx);
     size_t collection_count = get_collection_count();
+
     /* Any caller should expect at least one collection to exist. */
     testutil_assert(collection_count != 0);
-    return (get_collection(
```

Comment:
Is there a reason to prefer `std::advance` over `get_collection` here? 
The call to `_collections.erase(coll.id);` in `remove_random_collection` above implies we still depend on coll.id being identical to the collections position in `__collections`

Reply:
I am afraid of a race with the previous approach:

- `get_collection_count` holds a lock and then releases it.
- The assert is executed --> no lock here <--
- Then `get_collection` holds a lock again

During the step 2 we could delete a collection in parallel.

With the new approach, we have a lock during the whole time spent in `get_random_collection`. Does that make sense?

Reply:
Yep makes sense. Thanks for the clarification!